quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/KeycloakRecorder.java

```diff
@@ -215,4 +224,58 @@ public void setCryptoProvider(FipsMode fipsMode) {
     public void configureProtoStreamSchemas(List<SerializationContextInitializer> schemas) {
         Marshalling.setSchemas(schemas);
     }
+
+    public static void throwConfigError(String msg) {
+        // Ignore queued TRACE and DEBUG messages for not initialized log handlers
+        InitialConfigurator.DELAYED_HANDLER.setBuildTimeHandlers(new java.util.logging.Handler[]{});
+        throw new ConfigurationException(msg);
+    }
+
+    /**
+     * Print details about enabled multiple datasources and assert validations
+     */
+    public void manageMultipleDatasources(DataSourcesBuildTimeConfig dataSourcesConfig) {
+        Predicate<String> isDatasourceEnabled = ds -> DatabaseOptions.getResultNamedKey(DatabaseOptions.DB_ENABLED_DATASOURCE, ds)
+                .map(Configuration::isKcPropertyTrue)
+                .orElse(DatabaseOptions.DB_ENABLED_DATASOURCE.getDefaultValue().orElse(true));
+
+        List<String> enabledDatasources = dataSourcesConfig.dataSources().keySet().stream()
+                .filter(StringUtil::isNotBlank)
+                .filter(ds -> ds.equals("<default>") || isDatasourceEnabled.test(ds))
```

Comment:
We consider the default datasource enabled by default. Not sure if we need to check even that as IMO there will always be some default datasource.

Reply:
@mabartos the usage of enabled / active seems quite nuanced. Keycloak will be augmented to use the given source type(s), and specific named sources, but at runtime they are marked as inactive. 

Since we determine what provider / features are in use only at build-time - it seems like it should be taken care of at build-time, not run time.

Reply:
Yes, we should distinguish between enabled and active datasource as we are aware of used additional datasources in the build time. Users may deactivate the datasource in runtime. Updated.